server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -443,28 +466,42 @@ private DocumentMapper newDocumentMapper(Mapping mapping, MergeReason reason) {
             }
         }
 
-        Map<String, Mapping> results = new LinkedHashMap<>(2);
+        return new Mappings(defaultMapping, incomingMapping);
+    }
+
+    private static class Mappings {
```

Comment:
nice

Reply:
I was not sure if this was nice or horrible :) but it looked better than a map with two entries...